### ISSUE TITLE
Sanity tests failing in community.mongodb

### ISSUE THREAD

**Issue by dmsimard** [ansible-collections/community.mongodb#469]
##### SUMMARY
We are running sanity tests across every collection included in the Ansible community package (as part of [this issue](https://github.com/ansible-community/community-topics/issues/96)) and found that ``ansible-test sanity --docker`` against community.mongodb 1.3.4 fails with ansible-core 2.13.0rc1 in ansible 6.0.0a2.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
n/a

##### ANSIBLE VERSION

```paste below
ansible [core 2.13.0rc1]
```

##### COLLECTION VERSION

```paste below
1.3.4
```

##### STEPS TO REPRODUCE

```yaml
ansible-test sanity --docker
```

##### EXPECTED RESULTS

Tests are either passing or ignored.

##### ACTUAL RESULTS

```paste below
ERROR: Found 18 import issue(s) on python 3.10 which need to be resolved:
ERROR: plugins/module_utils/mongodb_common.py:6:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
ERROR: plugins/modules/mongodb_balancer.py:146:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_index.py:261:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_info.py:109:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_maintenance.py:65:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_monitoring.py:80:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_oplog.py:84:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
ERROR: plugins/modules/mongodb_parameter.py:79:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_replicaset.py:291:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_schema.py:160:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_shard.py:125:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_shard_tag.py:90:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_shard_zone.py:105:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_shell.py:198:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_shutdown.py:69:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_status.py:119:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_stepdown.py:95:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: plugins/modules/mongodb_user.py:198:0: traceback: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives (at plugins/module_utils/mongodb_common.py:6:0)
ERROR: The 1 sanity test(s) listed below (out of 43) failed. See error output above for details.
import --python 3.10
ERROR: Command "podman exec ansible-test-controller-DwgeTRTq /usr/bin/env ANSIBLE_TEST_CONTENT_ROOT=/root/ansible_collections/community/mongodb LC_ALL=en_US.UTF-8 /usr/bin/python3.10 /root/ansible/bin/ansible-test sanity --containers '{}' --skip-test pylint --metadata tests/output/.tmp/metadata-aackw0nr.json --truncate 0 --color no --host-path tests/output/.tmp/host-tmbhumt5" returned exit status 1.
```

**Comment by rhysmeister** [comment]
Can probably use https://packaging.pypa.io/en/latest/version.html#packaging.version.parse

**Comment by rhysmeister** [comment]
Done and merged to master.

**Comment by mariolenz** [comment]
@rhysmeister Thanks for the quick response!

FYI: We're running the sanity tests against the release included in the Ansible community package. It would be really great if you could release a new version containing your fix in time for 6.0.0a3 (Tuesday or Wednesday next week, I think). Otherwise, our tests will still fail for this collection.

**Comment by rhysmeister** [comment]
Done: https://github.com/ansible-collections/community.mongodb/pull/473
